compiler-core/src/language_server/engine.rs

```diff
@@ -476,10 +478,29 @@ fn hover_for_pattern(pattern: &TypedPattern, line_numbers: LineNumbers) -> Hover
 
 fn hover_for_module_statement(
     statement: &Definition<Arc<Type>, TypedExpr, SmolStr, SmolStr>,
+    byte_index: u32,
     line_numbers: LineNumbers,
 ) -> Option<Hover> {
     match statement {
         Definition::Function(fun) => {
+            if !fun.location.contains(byte_index) {
+                // `fun.location` is just the head of the function, so this ensures that
+                // hovering over the body doesn't produce any results.
+                return None;
+            }
+
+            // Show the type of the hovered argument to the user
+            for arg in &fun.arguments {
+                if arg.location.contains(byte_index) {
+                    let contents = Printer::new().pretty_print(&arg.type_, 0);
+                    return Some(Hover {
+                        contents: HoverContents::Scalar(MarkedString::String(contents)),
+                        range: Some(src_span_to_lsp_range(arg.location, &line_numbers)),
+                    });
+                }
+            }
```

Comment:
This finding logic should live with the other finding logic in `node_at_position`. This function is only about showing the information rather than hunting through the AST.

Reply:
I think that would involve extending the `Located` enum, right? I can have a go at that in the next few days (and also try to write some tests 🤞)